src/main.cpp

```diff
@@ -349,6 +349,22 @@ int main(int argc, char *argv[])
                FILENAMES["user_dir"].c_str());
         exit(1);
     }
+    if (!assure_dir_exist(FILENAMES["user_datadir"].c_str())) {
```

Comment:
`assure_dir_exist` accepts a `std::string`, there is no need to call `.c_str()` here.